### PR TITLE
Update OWNERS file to use approvers

### PR DESCRIPTION
**Pull request by spiffxp**
ref https://github.com/kubernetes/helm/issues/3333

Not sure our automation has ever used a field called "maintainers"

Base approvers on existing github teams that have write or admin
access to this repo, plus SIG leads

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -1,16 +1,9 @@
...
-  - prydonius
-  - seh
-  - technosophos
-  - thomastaylor312
-  - vaikas-google
-  - viglesiasce
+# See the OWNERS docs: https://git.k8s.io/community/docs/devel/owners.md
+
+approvers:
+  - sig-apps-leads
```
Comment (bacongobbler) [inline OWNERS]: Can we remove this alias and the alias from OWNERS_ALIASES? leaders from sig-apps do not have the right to approve helm PRs at this time. We just have two groups: core maintainers and reviewers.

I will note that this may change in the future with https://github.com/kubernetes/helm/issues/2792, but at this time SIG Apps leads do not have direct approval rights. We just report upstream to the weekly SIG Apps meetings.

On another note, it might be better to reference SIG Apps' OWNERS file here so that we get up-to-date changes when SIG Apps Leadership changes over time. That way there's less code duplication overall.

Thanks!

File: OWNERS_ALIASES [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# See the OWNERS docs: https://git.k8s.io/community/docs/devel/owners.md
+
+aliases:
+  sig-apps-leads:
+    - michelleN
+    - mattfarina
+    - prydonius
+    - kow3ns
+  helm-admins:
+    - sgoings
```
Comment (bacongobbler) [inline OWNERS_ALIASES]: Please remove. As awesome as @sgoings is, he is not a core maintainer and therefore should not be pinged on issues/PRs. Let's just remove admin rights from non-maintainers instead. :)

File: OWNERS_ALIASES [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    - technosophos
+    - vaikas-google
+  helm-maintainers:
+    - technosophos
+    - sgoings
```
Comment (bacongobbler) [inline OWNERS_ALIASES]: same, please remove. :)

File: OWNERS_ALIASES [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    - sparkprime
```
Comment (bacongobbler) [inline OWNERS_ALIASES]: same, please remove. :)

File: OWNERS_ALIASES [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    - michelleN
+    - sebgoa
+    - adamreese
+    - bmelville
```
Comment (bacongobbler) [inline OWNERS_ALIASES]: same, please remove. :)

File: OWNERS_ALIASES [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  helm-admins:
+    - sgoings
+    - technosophos
+    - vaikas-google
+  helm-maintainers:
+    - technosophos
+    - sgoings
+    - sparkprime
+    - michelleN
+    - sebgoa
```
Comment (bacongobbler) [inline OWNERS_ALIASES]: same, please remove. :)

**Comment by bacongobbler** [comment]
> Not sure our automation has ever used a field called "maintainers"

Up until this point, Helm's OWNERS file has never been used for automation. It's been there since Helm v2.3.0 (March 2017) as the process we use to note who's a core maintainer/reviewer and who's not. :)

I have a few inbound comments. Hold onto your butts!

**Review by bacongobbler — CHANGES_REQUESTED** [review]
For background context on my comments, here's a bit of a read on Helm's current review process: https://github.com/kubernetes/helm/issues/2792

File: OWNERS (hunk 1 above)
Comment (spiffxp) [reply OWNERS]: >is there a way we can reference a SIG Apps' OWNERS file here instead 

DRY teams isn't a today thing.  But on my radar.

File: OWNERS_ALIASES (hunk 2 above)
Comment (spiffxp) [reply OWNERS_ALIASES]: @bacongobbler should he then be removed from the @kubernetes/helm-admins team? the way this repo's collaborators are setup, that team currently grants him admin rights to this repo

File: OWNERS_ALIASES (hunk 3 above)
Comment (spiffxp) [reply OWNERS_ALIASES]: ditto, remove from @kubernetes/helm-maintainers ?

**Comment by spiffxp** [comment]
@bacongobbler maybe I should add you to the appropriate github teams and have you remove folks as appropriate? I'd rather be in the business of automating the way things are done today rather than administering your teams for you

File: OWNERS (hunk 1 above)
Comment (mattfarina) [reply OWNERS]: SIG Apps leads should not be in the owners files. Per the [community membership process](https://github.com/kubernetes/community/blob/master/community-membership.md), one needs to earn the right to have approver level and the leads have not done that by their role as lead.

Can we just remove the idea of `sig-apps-leads` from this whole PR?

**Review by mattfarina — CHANGES_REQUESTED** [review]
Can we simply swap out the `maintainers` section for an `approvers` section? A one line change? Possibly add a comment noting that this group is the maintainers.

I have a feeling that the groups managing access need a little tender loving care. I know we need to do that for charts as well.

**Comment by spiffxp** [comment]
My bad, I completely missed the collaborators on this repo and fired off this PR too early, will update shortly.  I am interested in making it easier to become a member and would be happy to sponsor any of these folks. (ref: https://github.com/kubernetes/community/issues/1565)

**Comment by cblecker** [comment]
I would also be happy to sponsor any of the helm maintainers for kubernetes org membership.

**Comment by bacongobbler** [comment]
I would rather we push back on this process until we suss out why there's a need to enforce all repos under the kubernetes org to use kubernetes/test-infra's tooling. To be honest I'm getting pretty frustrated with all of the processes being pushed onto the smaller kubernetes org communities.

A lot of the tooling that's been pushed onto Helm in the past few months (reviewable.io, SIG labels on github issues) have been largely irrelevant for the Helm community. We use github reviews and our own issue triaging system that has worked quite well for a project this size... We don't need or use more than half of the labels test-infra has forced onto the repository in recent weeks. Additionally, the "stale issue" bot that's been enforced across the org recently has done WAY more harm than good in terms of trust from community members. We use github issues to track meta issues, open bugs, proposals etc. All of which the Kubernetes org has a separate system (KEPs ,to name one). The stale issue bot may be fine for k8s, but it's doing more harm than good for Helm.

Additionally, we have some processes in place which the Kubernetes org has no replacement for at this time, like the recently announced [emeritus maintainer status](http://technosophos.com/2018/01/11/introducing-helm-emeritus-core-maintainers.html)) for core maintainers. By moving to Kubernetes' community process, where do all of our existing processes fit in?

In other words, can we *please* opt out of this... All of this? It doesn't make sense to push "big community" processes onto "little community" processes.

**Comment by spiffxp** [comment]
/close
Hi @bacongobbler @technosophos I apologize if my PR came across as a forced decision.  I like to  open PR's as places to have open discussion over an implementation of an idea.  I hear you loud and clear, and will close this.

My offer still stands to sponsor folks as org members so that you may administer the github teams that have write/admin access to your repo today.
